src/itest/java/uk/gov/companieshouse/company/profile/repository/RepositoryITest.java

```diff
@@ -68,4 +77,17 @@ void should_return_empty_optional_when_company_record_does_not_exist() {
                 this.companyProfileRepository.findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber("othernumber").isEmpty());
     }
 
+    @Test
+    void should_return_company_profile_based_on_company_number() {
+        CompanyProfile companyProfile = new CompanyProfile();
+        Data companyData = new Data().companyNumber(MOCK_COMPANY_NUMBER);
+        companyProfile.setData(companyData);
+        CompanyProfileDao companyProfileDao = new CompanyProfileDao(companyProfile);
+
+        this.companyProfileRepository.save(companyProfileDao);
+
+        CompanyProfileDao companyProfileRetrievedFromDB = this.companyProfileRepository.findByCompanyNumber(MOCK_COMPANY_NUMBER);
```

Comment:
Can reuse the existing query - `this.companyProfileRepository.findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber`